Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceDownsampleDepthLinear.azsl

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/Features/PostProcessing/FullscreenVertex.azsli>
+#include <Atom/Features/SrgSemantics.azsli>
+
+ShaderResourceGroup PassSrg : SRG_PerPass
+{
+    Texture2D<float2> m_input;
+
+    float m_invOutputScale;
+    uint  m_mipLevel;
+    bool  m_halfResolution;
+    uint  m_previousMipLevel;
+    uint  m_previousMipWidth;
+    uint  m_previousMipHeight;
+
+    Sampler PointSampler
+    {
+        MinFilter = Point;
+        MagFilter = Point;
+        MipFilter = Point;
+        AddressU  = Clamp;
+        AddressV  = Clamp;
+        AddressW  = Clamp;
+    };
+}
+
+struct PSOutput
+{    
+    float2 m_result : SV_Target0;
+};
+
+PSOutput MainPS(VSOutput IN)
+{
+    uint2 coords = IN.m_position.xy;
+    uint2 previousMipBaseCoords = IN.m_position.xy * 2;
+
+    float minDepth = 1e27f;
```

Comment:
This function needs comments related to it's purpose and any other specific details